lib/core/src/persist/mod.rs

```diff
@@ -104,20 +106,21 @@ impl Persister {
     pub(crate) fn insert_or_update_payment_with_wallet_tx(&self, tx: &WalletTx) -> Result<()> {
         let tx_id = tx.txid.to_string();
         let is_tx_confirmed = tx.height.is_some();
-        let amount_sat = tx
-            .balance
-            .iter()
-            .filter_map(|(asset_id, balance)| {
-                if *asset_id == lwk_wollet::elements::AssetId::LIQUID_BTC {
-                    return Some(balance);
-                }
-                None
-            })
-            .sum::<i64>();
-        if amount_sat == 0 {
-            log::warn!("Attempted to persist a payment with no output amount: tx_id {tx_id}");
-            return Ok(());
+        let mut tx_balances = tx.balance.clone();
+        // Remove the Liquid Bitcoin asset balance
+        let mut balance = tx_balances
+            .remove(&AssetId::LIQUID_BTC)
+            .map(|balance| (LIQUID_BTC_ASSET_ID.to_string(), balance));
```

Comment:
as far as I understand the balance in case of liquid bitcoin includes the fee. for example if we send 1000 sats and require to pay 40 sats fee then the balance is going to be 1040 and fee_sat=40.
How is the balance consisted for different assets? I am asking because the fees seems to be in liquid bitcoin (sats).

Reply:
Yeah, the balance for the L-BTC asset would be in this case -1040
When sending an asset, the balance (which is a hashmap of assets/balances) would be:
L-BTC: -40 (fee)
Asset: -1000 (asset amount)